File: frontend/keystone/src/services/workOrderService.ts
import axiosInstance from "../api/axiosConfig";
import { API } from "../constants/api";

import type { WorkOrder } from "../types/WorkOrder";
import type { AssignTechnicianRequest } from "../types/AssignTechnician";

class WorkOrderService {

    async getAllWorkOrders(): Promise<WorkOrder[]> {

        const response = await axiosInstance.get<WorkOrder[]>(
            API.WORKORDERS
        );

        return response.data;

    }

    async getWorkOrder(id: number): Promise<WorkOrder> {

        const response = await axiosInstance.get<WorkOrder>(
            `${API.WORKORDERS}/${id}`
        );

        return response.data;

    }

    async createWorkOrder(
        workOrder: WorkOrder
    ): Promise<WorkOrder> {

        const response = await axiosInstance.post<WorkOrder>(
            API.WORKORDERS,
            workOrder
        );

        return response.data;

    }

    async updateWorkOrder(
        id: number,
        workOrder: WorkOrder
    ): Promise<WorkOrder> {

        const response = await axiosInstance.put<WorkOrder>(
            `${API.WORKORDERS}/${id}`,
            workOrder
        );

        return response.data;

    }

    async assignTechnician(
        id: number,
        request: AssignTechnicianRequest
    ): Promise<WorkOrder> {

        const response = await axiosInstance.put<WorkOrder>(
            `${API.WORKORDERS}/${id}/assign`,
            request
        );

        return response.data;

    }

}

export default new WorkOrderService();

File: backend/deliverableservice/src/main/java/com/keystone/deliverableservice/service/WorkOrderService.java
package com.keystone.deliverableservice.service;

import java.util.List;

import com.keystone.deliverableservice.dto.request.AssignTechnicianRequest;
import com.keystone.deliverableservice.dto.request.WorkOrderRequest;
import com.keystone.deliverableservice.dto.response.WorkOrderResponse;

public interface WorkOrderService 
{
	WorkOrderResponse createWorkOrder(WorkOrderRequest request);
	WorkOrderResponse updateWorkOrder(Long id, WorkOrderRequest request);
	
	WorkOrderResponse getWorkOrder(Long id);
	
	List<WorkOrderResponse> getAllWorkOrders();
	
	WorkOrderResponse assignTechnician(Long workOrderId, AssignTechnicianRequest request);
	
	void deleteWorkOrder(Long id);

}
